### PR TITLE
DAOS-2070 vos: Fix a PMDK issue with btree test

### PR DESCRIPTION
**Pull request by jolivier23**

### CODE REVIEW

File: src/common/tests/btree.c [hunk 1]
```diff
@@ -126,7 +126,6 @@ ik_rec_free(struct btr_instance *tins, struct btr_record *rec, void *args)
 		umem_id_t *rec_ret = (umem_id_t *) args;
 		 /** Provide the buffer to user */
 		*rec_ret	= rec->rec_mmid;
-		rec->rec_mmid	= UMMID_NULL;
```
Comment (Nasf-Fan) [inline src/common/tests/btree.c]: Is it redundant or any side-effect? As my understand, if the caller offer the @args parameter, then it needs to free the space (pointed by rec->rec_mmid) by itself. The @rec will be freed after ik_rec_free() returned. So whether reset the rec_mmid or not will not cause correctness issue, right?

Comment (jolivier23) [reply src/common/tests/btree.c]: Right, the caller is responsible to free the parameter as returned.

Comment (Nasf-Fan) [reply src/common/tests/btree.c]: Then what is the side-effect if we keep this line? Just redundant or some some potential wrong behavior?

Comment (NiuYawei) [reply src/common/tests/btree.c]: It seems to me that it's redundant, and it probably may cause warning in some code checking tool?

Comment (jolivier23) [reply src/common/tests/btree.c]: It causes valgrind's pmemcheck tool to report a write to persistent memory that isn't part of the transaction

**Pull request merged**